config-templates/redhat/odf-remote/4.10/custom-parameters.json

```diff
@@ -4,21 +4,27 @@
     "displayname": "OSD pod volume size",
     "name": "osd-size",
     "default": "100Gi",
+    "regex": "^[1-9]{1}[0-9]{0,3}(?:Gi|Ti)$",
+    "min-length": "3",
+    "max-length": "6",
     "place-holder": "100Gi",
-    "regex": "^[0-9]+Gi$",
     "required": "true"
   },
   {
     "description": "The storage class to use when dynamically provisioning volumes for the OSD pods.",
     "displayname": "OSD pod storage class",
     "name": "osd-storage-class",
+    "max-length": "43",
```

Comment:
@anandkarra Please validate if this is an user input field and not a dropdown, if it is a field where user can enter any name then we should have validation around minimum length as well?

Reply:
This is a user input field.

If the length exceeds 63 characters this is the error we get:

```
failed to create osd for device set "ocs-deviceset-ibmc-vpc-block-metro-retain-general-purpose-1" for count 0. failed to create pvc "" for device set "ocs-deviceset-ibmc-vpc-block-metro-retain-general-purpose-1": PersistentVolumeClaim "ocs-deviceset-ibmc-vpc-block-metro-retain-general-purpose-sdqk8" is invalid: metadata.labels: Invalid value: "ocs-deviceset-ibmc-vpc-block-metro-retain-general-purpose-1-data-0": must be no more than 63 characters
```

Here, the prefix is `ocs-deviceset-` and suffix is `-1-data-0`; a total of 23 characters. This leaves 40 characters for the user input.